polar_route/crossing_smoothing.py

```diff
@@ -0,0 +1,1546 @@
+            apply waypoint_correction to get path related information along the path
+
+            Inputs:
+                adjacent_pairs (list of type find_edge) - A list of the adjacent cell pairs in the form of find_edge
+                start_waypoint (tuple) - Start waypoint (long,lat)
+                end_waypoint (tuple) - End waypoint (long,lat)
+
+        '''
+        # Initialising zero arrays for the path variables 
+        variables =  {}    
+        for var in self.path_requested_variables:
+            variables[var] ={}
+            variables[var]['path_values'] = np.zeros(len(adjacent_pairs)+2)
+
+
+        # Path point
+        path_points = [start_waypoint]
+
+        # Looping over the path and determining the variable information
+        for ii in range(len(adjacent_pairs)+1):
+            if ii == 0:
+                Wp = start_waypoint
+                Cp = adjacent_pairs[ii].crossing
+                cellbox = adjacent_pairs[ii].start
+            elif ii == (len(adjacent_pairs)):
+                Wp = adjacent_pairs[ii-1].crossing
+                Cp = end_waypoint
+                cellbox = adjacent_pairs[ii-1].end
+            else:
+                Wp = adjacent_pairs[ii-1].crossing
+                Cp = adjacent_pairs[ii].crossing
+                cellbox = adjacent_pairs[ii].start
+
+            # Adding End point
+            path_points += [Cp]
+
+            # Determining the value for the variable for the segment of the path and the corresponding case
+            segment_variable, segment_case = self._waypoint_correction(self.path_requested_variables,cellbox,Wp,Cp)
+
+            # Adding that value for the segment along the paths
+            for var in segment_variable:
+                if type(segment_variable[var]) == np.ndarray:
+                    variables[var]['path_values'][ii+1] = segment_variable[var][segment_case]
+                else:
+                    variables[var]['path_values'][ii+1] = segment_variable[var]
+
+
+        # Applying processing to all path values
+        for var in variables.keys():
+            processing_type = self.path_requested_variables[var]['processing']
+            if type(processing_type) == type(None):
+                continue
+            elif processing_type == 'cumsum':
+                variables[var]['path_values'] = np.cumsum(variables[var]['path_values'])
+
+
+        path_info = {}
+        path_info['path']      = np.array(path_points)
+        path_info['variables'] = variables
+
+        return path_info
+
+#======================================================
+class Smoothing:
+    def __init__(self,dijkstra_graph,adjacent_pairs,start_waypoint,end_waypoint,blocked_metric='SIC'):
+        '''
+            Class construct that has all the operations requried for path smoothing. Including: Relationship of adjacent pairs,
+            edge finding new edges to add and returns a list of the adjacent pairs for the constructed path
```

Comment:
It would be good if you could describe the arguments to the init method in this docstring.